### ISSUE TITLE
How to set setDynamicRange right? No change on inference time, but setInt8Calibrator faster too much.

### ISSUE THREAD

**Issue by doomooo** [NVIDIA/TensorRT#2031]
## Description
Hi, I wander how to set setDynamicRange right? No change on inference time, but setInt8Calibrator faster too much.

## Environment

**TensorRT Version**:  723
**NVIDIA GPU**: 3080
**NVIDIA Driver Version**: 
**CUDA Version**: 
**CUDNN Version**: 
**Operating System**: 
**Python Version (if applicable)**: 
**Tensorflow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 
**Baremetal or Container (if so, version)**:

**Comment by nvpohanh** [comment]
@doomooo Could you check if you have called setDynamicRange() for ALL intermediate tensors? If a tensor does not have dynamic range set, that layer won't run in INT8

**Comment by doomooo** [comment]
Thanks for your reply.
"ALL intermediate tensors" means all networks' tensors? or specified layers' tensors. I just set all conv layers' outputs.  (like [sampleINT8API](https://github.com/NVIDIA/TensorRT/blob/release/7.2/samples/opensource/sampleINT8API/sampleINT8API.cpp))
And, if I extract QAT's params and set it in setInt8Calibrator instead of DynamicRange , any risk? 
@nvpohanh

**Comment by nvpohanh** [comment]
> "ALL intermediate tensors" means all networks' tensors? or specified layers' tensors. I just set all conv layers' outputs.

I meant the input/output tensors for **each** layer, including Conv, Scale, ReLU, etc.

> And, if I extract QAT's params and set it in setInt8Calibrator instead of DynamicRange , any risk or problem?

That should work.

**Comment by nvpohanh** [comment]
Note that if the ONNX file contains the Q/DQ nodes, you can use TRT 8.2 directly with the ONNX file without the need to set the dynamic ranges.

**Comment by doomooo** [comment]
> I meant the input/output tensors for **each layer**, including Conv, Scale, ReLU, etc.

Why is "each layer", can I set int8 only for the first half layers?

Thanks for your notes. I used TRT8.2 successfully. But due to some limitations, I should use TRT7.2, many problems show when I use QAT ONNX with TRT7.2 directly. So, I set int8 params manually.

**Comment by doomooo** [comment]
@nvpohanh 

> I meant the input/output tensors for each layer, including Conv, Scale, ReLU, etc.

Why is "each layer", can I set int8 only for the first half layers?

Thanks for your notes. I used TRT8.2 successfully. But due to some limitations, I should use TRT7.2, many problems show when I use QAT ONNX with TRT7.2 directly. So, I set int8 params manually.

**Comment by nvpohanh** [comment]
> Why is "each layer", can I set int8 only for the first half layers?

You can set dynamic ranges on only some layers, but the other layers won't be able to use INT8 tactics, so the performance will be worse than the case when you use calibrator.

**Comment by doomooo** [comment]
Thanks, I got it.  @nvpohanh 
From your description before. I think that maybe I ignore ` [Scale]_output`.  
And, what is ` [Scale]_output`. used for?

**Comment by nvpohanh** [comment]
The Scale layers are for BatchNorm ops. If you check the order of the layers in a network, you will see the Scale layers sitting between the Conv and the ReLU layers.

**Comment by doomooo** [comment]
Thanks, If I have absorbed and merged it into Conv in Tensorflow, will influence the int8 inference here?

**Comment by doomooo** [comment]
Thanks, If I have absorbed and merged it into Conv in Tensorflow, will influence the int8 inference here? @nvpohanh

**Comment by nvpohanh** [comment]
Maybe not. But it depends on how the ops are described in ONNX file, so the ONNX file is still the golden reference.

**Comment by doomooo** [comment]
Thank you for your kindly reply. @nvpohanh
